pkg/virt-controller/services/renderresources.go

```diff
@@ -18,9 +18,10 @@ type ResourceRenderer struct {
 	vmRequests         k8sv1.ResourceList
 	calculatedLimits   k8sv1.ResourceList
 	calculatedRequests k8sv1.ResourceList
+	extraResources     k8sv1.ResourceList
 }
 
-func NewResourceRenderer(vmLimits k8sv1.ResourceList, vmRequests k8sv1.ResourceList, options ...ResourceRendererOption) *ResourceRenderer {
+func NewResourceRenderer(vmLimits k8sv1.ResourceList, vmRequests k8sv1.ResourceList, extraResources k8sv1.ResourceList, options ...ResourceRendererOption) *ResourceRenderer {
```

Comment:
In hindsight, I think you're right ... Feels clunky to have the tun / kvm devices as a "special" resource type. 

My motivation was simple ... It is **very** probable they will be used - hence, I was passing them via the constructor. 

... but definitely feels clunky. I'll re-think this through, see if I can come up w/ an alternative. Otherwise, having them as just another option, is also simpler to reason about that what I currently have.